src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -319,6 +330,9 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
       }
     )
 
+  override def scanRight[B](z: B)(op: (A, B) => B): LazyList[B] =
+    reverse.scanLeft(z)((b, a) => op(a, b))
```

Comment:
We should probably replace the default implementation of `scanRight` with this one, wdyt?

Reply:
Perhaps. I think there are probably some situations where the current implementation is better than this one (perhaps for `List`? Some other `LinearSeq`s? I'm not sure). At the same time, neither work great for `IndexedSeq`, which seems less than ideal.

I would be okay with making this the default implementation, but I think it might even be worth looking for a better one. For instance, would `iterableFactory.from(reverseIterator.scanLeft(...)(...))` be a better default? It seems like it would perform reasonably well for `IndexedSeq`.

Reply:
(for reference, the current implementation is below. `reversed` is overridden sensibly in `IndexedSeqOps`)

```scala
  def scanRight[B](z: B)(op: (A, B) => B): CC[B] = {
    var scanned = z :: immutable.Nil
    var acc = z
    for (x <- reversed) {
      acc = op(x, acc)
      scanned ::= acc
    }
    iterableFactory.from(scanned)
  }

  protected def reversed: Iterable[A] = {
    var xs: immutable.List[A] = immutable.Nil
    val it = iterator
    while (it.hasNext) xs = it.next() :: xs
    xs
  }
```

Reply:
> would `iterableFactory.from(reverseIterator.scanLeft(...)(...))` be a better default?

I agree!

Reply:
I've actually checked, and this implementation of `scanRight` for `LazyList` is wrong - it puts the elements in reverse order. It needs another `reverse` at the end, at which point I'm unsure of the benefits of it as a default implementation.

Reply:
Yes. It just calls `reverse.iterator`

Reply:
To have the elements in the correct order, you would need `iterableFactory.from(reverseIterator.scanLeft(...)(...)).reverse`, which definitely doesn't seem efficient

Reply:
It does seem to me that for some `IndexedSeq`s (e.g. `ArraySeq`) you could have a really efficient implementation that creates a structure one element larger than the current one, and then populates the indices in reverse order. I think that's out of the scope of this PR however.